### PR TITLE
[MNT] Minor neuralforecast changes

### PR DESCRIPTION
**Pull request by yarnabrina**
This PR is pure refactoring with no logic change.

1. created type alias for supported scaling methods
2. updated documented link to all optimisers instead of just Adam
3. added @geetu040 and @pranavvp16 in author section of the adapter
4. simplified `auto=True` logic by reducing nesting (requesting @geetu040 for a review)
5. changed variable names for readability and avoided unnecessary variables and conversions (requesting @pranavvp16 to take a look)

### CODE REVIEW

**Comment by yarnabrina** [comment]
As informed in Discord, both `main` branch and this PR fails with frequence issues with pandas 2.2 alias changes. However, that seemed out of scope for this PR.

**Comment by pranavvp16** [comment]
LGTM! for the pandas failure I have noticed it fails of `freq=auto` previously, as try to interpret frequency using the `ForecastingHorizon ` class in sktime

File: sktime/forecasting/base/adapters/_neuralforecast.py [hunk 1]
```diff
@@ -137,66 +139,73 @@ def algorithm_parameters(self: "_NeuralForecastAdapter") -> dict:
         - custom model name (``alias``) - used from ``algorithm_name``
         """
 
-    def _get_valid_parameters(self: "_NeuralForecastAdapter") -> dict:
-        """Get valid parameters for the underlying NeuralForecast algorithm class.
+    def _ignore_invalid_parameters(self: "_NeuralForecastAdapter") -> dict:
```
Comment (VascoSch92) [inline sktime/forecasting/base/adapters/_neuralforecast.py:142]: Why don't you use Dict from typing?

Comment (yarnabrina) [reply sktime/forecasting/base/adapters/_neuralforecast.py:142]: `Dict` is deprecated and `dict` is recommended.

https://docs.python.org/3/library/typing.html#typing.Dict

**Comment by geetu040** [comment]
> 4\. simplified `auto=True` logic by reducing nesting (requesting @geetu040 for a review)

All okay from my side, changes appear to be all good, covering all the code branches.

**Review by fkiraly — APPROVED** [review]

**Pull request merged**